src/matrixeventprocessor.ts

```diff
@@ -367,6 +360,36 @@ export class MatrixEventProcessor {
         return embed;
     }
 
+    private async GetUserProfileForRoom(roomId: string, userId: string) {
+        const mxClient = this.bridge.getClientFactory().getClientAs();
+        const intent = this.bridge.getIntent();
+        let profile: {displayname: string, avatar_url: string} | undefined;
+        try {
+            profile = this.mxUserProfileCache.get(`${roomId}:${userId}`);
+            if (!profile) {
+                log.verbose(`Profile ${userId}:${roomId} not cached`);
+                profile = await mxClient.getStateEvent(roomId, "m.room.member", userId);
+            }
+            if (profile) {
+                this.mxUserProfileCache.set(`${roomId}:${userId}`, profile);
```

Comment:
move cache setting into `if(!profile) {...}` and that other else condition then another `if(!profile){...}`? Else our timed cache doesn't matter as we keep on resetting the cache timeout